premerge/routine-maintenance.md

```diff
@@ -0,0 +1,116 @@
+to ensure things keep working smoothly and that we are not vulernable to
+security issues.
+
+### Getting Notified of Version Updates
+
+There are several pieces of software that we want to upgrade relatively
+quickly (like the Github Actions Runner binary). Because of that, knowing
+when a new version is released is important. The easiest way to do this is
+to subscribe to new release notifications on Github. If you go to a
+repository, you can click on the watch button, select custom, and then
+select releases. Any new releases for that repository will show up in
+your Github notifications.
+
+Releases from the following repositories generally require an update on the
+premerge infrastructure side:
+
+1. https://github.com/actions/actions-runner-controller
+2. https://github.com/actions/runner
+3. https://github.com/llvm/llvm-project
+
+### Github Actions Runner Binary
+
+The runner binary is what runs inside the containers on the cluster to
+execute jobs and report status results back to Github. The runner binary
+has a relatively short time horizon (about six months) before it becomes
+unsupported by Github and it will no longer work.
+
+When a new runner binary is released, there are three places that need to
+be updated in a PR against the LLVM monorepo:
+
+1. The Linux CI container - The `Dockerfile` at
+`.github/workflows/containers/github-action-ci/Dockerfile` has an environment
+variable towards the bottom of the file called `GITHUB_RUNNER_VERSION` that
+needs to be updated to the new version.
+2. The Windows CI container - The `Dockerfile` at
+`.github/workflows/containers/github-action-ci-windows/Dockerfile` has an
+argument called `RUNNER_VERSION` near the bottom of the file that needs to
+be updated to the new version.
+3. The libc++ CI container - The `docker-compose` manifest at
+`libcxx/utils/ci/docker-compose.yml` needs to be updated to pull in the latest
+runner images using the [libc++ instructions](https://libcxx.llvm.org/Contributing.html#updating-the-ci-testing-container-images)
+
+### Other Container Image Software
+
+The container images also contain many other pieces of software critical
+for building LLVM, like CMake, ninja, and the toolchain itself. Keeping
+most of these up to date is ideal.
+
+A large amount of the software comes from the distribution and thus does not
+need to be explicitly updated. We prefer to install software from the
+distribution when possible. However, this does mean that distribution
+updates are quite important. To update the distribution for the Linux container,
+perform the following steps:
+
+1. Modify `.github/workflows/containers/github-action-ci/Dockerfile` locally
+to use the latest `ubuntu:xx.04` image.
+2. Ensure that `monolithic-linux.sh` with all options enabled runs successfully.
+Push any needed changes.
+3. Push the updated `Dockerfile` and update the workflow in
+`.github/workflows/build-ci-container.yml` to use the correct image name.
```

Comment:
For #3: Can you be more explicit about what to update in build-ci-container.yml?  Do you mean the 'container_name" in the bulid-ci-container job? or something else? What's the right value to update it to?